src/os/bluestore/BlueFS.cc

```diff
@@ -2390,6 +2393,31 @@ int BlueFS::_flush_range(FileWriter *h, uint64_t offset, uint64_t length)
   }
   ceph_assert(bl.length() == length);
 
+  h->pos = offset + length;
+
+  unsigned tail = bl.length() & ~super.block_mask();
+  if (tail) {
+    dout(20) << __func__ << " caching tail of 0x"
+             << std::hex << tail
+             << " and padding block with 0x" << (super.block_size - tail)
+             << std::dec << dendl;
+    h->tail_block.substr_of(bl, bl.length() - tail, tail);
+    bl.append_zero(super.block_size - tail);
```

Comment:
original tail processing had different flows depending on ino. For the sake of performance I suppose. Any rationales for eliminating this behavior?

Reply:
No, it is an omission. THANKS.

Reply:
@ifed01 Current implementation of buffer::list::append_zero does it in effective way. Very similar to what was previously in non-zero ino case.